plugins/ffxiv/ffxiv.cpp

```diff
@@ -0,0 +1,203 @@
+	}
+
+	// Boolean value to check if game addresses retrieval is successful
+	bool ok;
+	// Create containers to stuff our raw data into, so we can convert it to Mumble's coordinate system
+	float avatar_pos_corrector[3], camera_pos_corrector[3], avatar_azimuth, camera_azimuth, camera_elevation;
+	// Values for extra features
+	char player[64];
+	unsigned int map_id;
+	// State
+	unsigned char state, camera_is_free;
+
+	// Avatar pointer
+	procptr32_t avatar_address = peekProc<procptr32_t>(pModule + avatar_ptr);
+	if (!avatar_address) return false;
+
+	// Camera pointer
+	procptr32_t camera_address = peekProc<procptr32_t>(pModule + camera_ptr);
+	if (!camera_address) return false;
+
+	// Peekproc and assign game addresses to our containers, so we can retrieve positional data
+	ok = peekProc(pModule + state_offset, &state, 1) && // Magical state value: 0 or 255 when in main menu and 1 when in-game.
+			peekProc(camera_address + camera_pos_offset, &camera_pos_corrector, 12)	 && // Camera Position values (X, Z and Y).
+			peekProc(camera_address + camera_azimuth_offset, &camera_azimuth, 4)     && // Camera azimuth float.
+			peekProc(camera_address + camera_elevation_offset, &camera_elevation, 4) && // Camera elevation float.
+			peekProc(camera_address + camera_is_free_offset, &camera_is_free, 1)	 && // Camera is in first person mode
+			peekProc(avatar_address + avatar_pos_offset, &avatar_pos_corrector, 12)	 && // Avatar Position values (X, Z and Y).
+			peekProc(avatar_address + avatar_azimuth_offset, &avatar_azimuth, 4)     && // Avatar azimuth float.
+			peekProc(pModule + map_id_offset, &map_id, 4)							 && // Map id.
+			peekProc(avatar_address + identity_offset, player);							// Player name.
+
+	// This prevents the plugin from linking to the game in case something goes wrong during values retrieval from memory addresses.
+	if (! ok)
+		return false;
+
+	// State
+	if (state != 1) { // If not in-game
+		context.clear(); // Clear context
+		identity.clear(); // Clear identity
+		// Set vectors values to 0.
+		for (int i=0;i<3;i++) {
+			avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] =  camera_front[i] = camera_top[i] = 0.0f;
+		}
+
+		return true; // This tells Mumble to ignore all vectors.
+	}
+
+	// Begin context
+	std::ostringstream ocontext;
+	ocontext << "{";
+
+	// Map id
+	ocontext  << "Map: " << map_id << ""; // Set map id in identity.
+
+	ocontext << "}";
+	context = ocontext.str();
+	// End context
+
+	// Begin identity
+	std::wostringstream oidentity;
+	oidentity << "{";
+
+	// Map id
+	oidentity << "Map: " << map_id << ", "; // Set map id in identity.
+
+	// Player name
+	escape(player, sizeof(player));
+	if (strcmp(player, "") != 0) {
+		oidentity << "Player: \"" << player << "\""; // Set player name in identity.
+	} else {
+		oidentity << "Player: null";
+	}
+
+	oidentity << "}";
+	identity = oidentity.str();
+	// End identity
+
+	/*
+	Mumble | Game (in L hand coordinate system)
+	X      | -X
```

Comment:
I have no opinion. As long as the coordinate system is mapped sensibly onto the Mumble coordinate system, I'm happy.